internal/service/streamprocessor/resource_test.go

```diff
@@ -0,0 +1,195 @@
+					resource.TestCheckResourceAttr(resourceName, "processor_name", processorName),
+					resource.TestCheckResourceAttr(resourceName, "state", "STOPPED"),
+				),
+			},
+		}})
+}
+
+func TestAccStreamProcessorRS_failWithInvalidStateOnCreation(t *testing.T) {
+	var (
+		projectID     = acc.ProjectIDExecution(t)
+		processorName = "new-processor"
+		instanceName  = acc.RandomName()
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acc.PreCheckBasic(t) },
+		ProtoV6ProviderFactories: acc.TestAccProviderV6Factories,
+		CheckDestroy:             checkDestroyStreamProcessor,
+		Steps: []resource.TestStep{
+			{
+				Config:      streamProcessorConfigWithSampleConnection(projectID, instanceName, processorName, "STOPPED"),
+				ExpectError: regexp.MustCompile("When creating a stream processor, the only valid states are CREATED and STARTED"),
+			},
+		}})
+}
+
+func streamProcessorConfigWithSampleConnection(projectID, instanceName, processorName, state string) string {
+	// Add mongodbatlas_stream_connection once sample stream connection is not created by default
+	// resource "mongodbatlas_stream_connection" "sample" {
+	// 	project_id      = %[1]q
+	// 	instance_name   = mongodbatlas_stream_instance.instance.instance_name
+	// 	connection_name = "sample_stream_solar_2"
+	// 	type            = "Sample"
+	// }
+	stateConfig := ""
+	if state != "" {
+		stateConfig = fmt.Sprintf(`state = %[1]q`, state)
+	}
+	return fmt.Sprintf(`
+	resource "mongodbatlas_stream_instance" "instance" {
+		project_id    = %[1]q
+		instance_name = %[2]q
+		data_process_region = {
+			region         = "VIRGINIA_USA"
+			cloud_provider = "AWS"
+		}
+	}
+
+	resource "mongodbatlas_stream_processor" "processor" {
+		project_id     = %[1]q
+		instance_name  = mongodbatlas_stream_instance.instance.instance_name
+		processor_name = %[3]q
+		pipeline       = "[{\"$source\":{\"connectionName\":\"sample_stream_solar\"}},{\"$emit\":{\"connectionName\":\"__testLog\"}}]"
+		%[4]s
+	}
+	`, projectID, instanceName, processorName, stateConfig)
+}
+
+func checkExists(resourceName string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+		projectID := rs.Primary.Attributes["project_id"]
+		instanceName := rs.Primary.Attributes["instance_name"]
+		processorName := rs.Primary.Attributes["processor_name"]
+		_, _, err := acc.ConnV2().StreamsApi.GetStreamProcessor(context.Background(), projectID, instanceName, processorName).Execute()
+
+		if err != nil {
+			return fmt.Errorf("Stream processor (%s) does not exist", processorName)
+		}
+
+		return nil
+	}
+}
+
+func checkDestroyStreamProcessor(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "mongodbatlas_stream_processor" {
+			continue
+		}
+		ids := conversion.DecodeStateID(rs.Primary.ID)
```

Comment:
I don't think we should use `DecodeStateID` here, should use same style as in `checkExists`?